### PR TITLE
refactor: pylint mypy low hanging

### PR DESCRIPTION
**Pull request by jdpatt**
Last one of the three like in pyibisami.  Anything beyond this should be altering features/adding features and will denote them as such.

- Just gets the low hanging from from running `tox -e format` and `tox -e lint` which clear up a large majority of pylint's complaints or increases the score of 7.4 to 8.76 out of 10.0.
- Updates mypy to work with the src/ structure which is run with `tox -e type-check` cleaned up an error or two and told it to skip simbeor.

### CODE REVIEW

**Comment by capn-freako** [comment]
Hi @jdpatt ,

Do you think that these conflicts are the result of my botched attempt to merge your PR 108?
Should we go back and redo that one before moving forward with this one?

File: MANIFEST.in [hunk 1]
```diff
@@ -1,10 +1,4 @@
 include README.md
 include LICENSE
-include requirements.txt
-include pybert/images/icon.png
-recursive-include docs/
-recursive-include tests/
-global-exclude *.swp
-global-exclude *.DS_Store
-global-exclude *.buildinfo
```
Comment (capn-freako) [inline MANIFEST.in:9]: Why did you remove these 3 global excludes?

File: README.md [hunk 2]
```diff
@@ -27,24 +27,15 @@ Notice: Before using this package for any purpose, you MUST read and understand
...
 
-## Testing
-
-Tox is used for the test runner and documentation builder.  By default, it runs the following
-environments: _py36_, _py37_, _pylint_, _flake8_ and _docs_.  It will skip any missing python versions.
-* `pip install tox`
-* `tox`
-
-To run a single environment such as "docs" run: `tox -e docs`
-
```
Comment (capn-freako) [inline README.md:38]: Why did you remove this section on testing?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,96 @@
+[build-system]
```
Comment (capn-freako) [inline pyproject.toml:1]: This one also already came in, with PR108.

File: requirements-dev.txt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+pytest-xdist
+pylint
+pre-commit
+mypy
+tox
+
+# Documentation tooling
+sphinx
+m2r
+
```
Comment (capn-freako) [inline requirements-dev.txt:18]: Everything above already came in, with #108 .
The remaining two lines appear to be new, however.

File: src/pybert/pybert.py [hunk 5]
```diff
@@ -14,31 +13,23 @@
 
 Copyright (c) 2014 by David Banas; All rights reserved World wide.
 """
-from traits.etsconfig.api import ETSConfig
-# ETSConfig.toolkit = 'qt4'  # Yields unacceptably small font sizes in plot axis labels.
-# ETSConfig.toolkit = 'qt4.celiagg'   # Yields unacceptably small font sizes in plot axis labels.
-# ETSConfig.toolkit = 'qt.qpainter'  # Was causing crash on Mac.
-# ETSConfig.toolkit = 'qt.image'     # Program runs, but very small fonts in plot titles and axis labels.
-# ETSConfig.toolkit = 'wx'           # Crashes on launch.
```
Comment (capn-freako) [inline src/pybert/pybert.py:22]: Would you restore these 6 lines, please?
I'm using them to experiment with, and provide feedback on, different GUI backend alternatives, which are under active current development.

File: src/pybert/pybert.py [hunk 6]
```diff
@@ -53,43 +44,38 @@
     Property,
     Range,
     String,
-    Trait,
+    cached_property,
 )
+from traits.etsconfig.api import ETSConfig
```
Comment (capn-freako) [inline src/pybert/pybert.py:49]: This needs to remain at the top of the import stack, because it's important that the (currently commented out) lines that use it execute before anything else.

File: src/pybert/pybert.py [hunk 7]
```diff
@@ -53,43 +44,38 @@
...
-    interp_s2p,
-    renorm_s2p,
+    trim_impulse,
 )
 
+# ETSConfig.toolkit = 'qt4'  # Yields unacceptably small font sizes in plot axis labels.
+# ETSConfig.toolkit = 'qt4.celiagg'   # Yields unacceptably small font sizes in plot axis labels.
+# ETSConfig.toolkit = 'qt.qpainter'  # Was causing crash on Mac.
+# ETSConfig.toolkit = 'qt.image'     # Program runs, but very small fonts in plot titles and axis labels.
+# ETSConfig.toolkit = 'wx'           # Crashes on launch.
```
Comment (capn-freako) [inline src/pybert/pybert.py:76]: These, when uncommented, _must_ execute before anything else.

File: src/pybert/pybert.py [hunk 8]
```diff
@@ -362,7 +347,7 @@ def __init__(self, name="(noname)", enabled=False, min_val=0.0, max_val=0.0, val
 
         # Super-class initialization is ABSOLUTELY NECESSARY, in order
         # to get all the Traits/UI machinery setup correctly.
-        super(TxTapTuner, self).__init__()
+        super().__init__()
```
Comment (capn-freako) [inline src/pybert/pybert.py:350]: Can you tell me what happened here?
Do you know why `TxTapTuner` and `self` were being passed originally?

File: src/pybert/pybert.py [hunk 9]
```diff
@@ -645,7 +638,7 @@ def __init__(self, run_simulation=True, gui=True):
 
         # Super-class initialization is ABSOLUTELY NECESSARY, in order
         # to get all the Traits/UI machinery setup correctly.
-        super(PyBERT, self).__init__()
+        super().__init__()
```
Comment (capn-freako) [inline src/pybert/pybert.py:641]: Same question as above.

File: src/pybert/solvers/simbeor/__init__.py [hunk 10]
```diff
@@ -31,9 +35,6 @@
 class Solver(slvr.Solver):
     """Simbeor-specific override of `pybert.solver.Solver` class."""
 
-    def __init__(self):
-        super().__init__()
-
```
Comment (capn-freako) [inline src/pybert/solvers/simbeor/__init__.py:36]: Superclass initialization no longer needs to be explicitly invoked?

File: tox.ini [hunk 11]
```diff
@@ -1,69 +1,59 @@
 [tox]
-envlist = py38, pylint, format, flake8, docs
+envlist = py38, py39, py310, pylint, format, flake8, docs
 skip_missing_interpreters = true
+isolated_build = true
 
 [testenv]
-changedir = tests
```
Comment (capn-freako) [inline tox.ini:6]: Why don't we want to do this anymore?

File: tox.ini [hunk 12]
```diff
@@ -1,69 +1,59 @@
...
-commands = 
-    pylint --rcfile={toxinidir}/tox.ini pybert
-
-[testenv:flake8]
-deps = 
-    flake8
-    numpy
-    Cython
-commands = 
-    flake8 pybert
```
Comment (capn-freako) [inline tox.ini:37]: Are these just being updated to: `lint` and `type-check`, respectively?

**Review by capn-freako — CHANGES_REQUESTED** [review]
Hi @jdpatt ,

Wow! That was a significant amount of work; thanks!
I do have some questions that I'd like to discuss with you before we merge.

Thanks,
-db

**Comment by jdpatt** [comment]
> Hi @jdpatt ,
> 
> Do you think that these conflicts are the result of my botched attempt to merge your PR 108? Should we go back and redo that one before moving forward with this one?
> 
> I guess that wasn't it, because I just:
> 
> * reset master to its state just before I squash-merged your PR108 locally (i.e. - on my laptop),
> * squash-merged your [chore: Update package tooling #108](https://github.com/capn-freako/PyBERT/pull/108) _on GitHub_, which appears to have resulted in a clean merge this time, and
> * reloaded this PR, which still reports conflicts. :(
> 
> Is it possible that there are conflicts between your PRs 108 and 109, which have nothing to do with my `master`?

I think its due to me breaking up PRs how I did.  I won't do that again and I will work in branches on your repo to prevent you having to deal with those headaches again!  Let me try to clean up the diff even if that means closing this PR and opening a new one while fixing your concerns.

File: tox.ini (hunk 12 above)
Comment (jdpatt) [reply tox.ini:37]: Correct! When I actually started to use them, I found a few more issues that needed altering.  The diff is terrible and I apologize.  Let me see if github can clear that up.

File: tox.ini (hunk 11 above)
Comment (jdpatt) [reply tox.ini:6]: I added tests/ to the end of the pytest commands.  When tox changes directories, it apparently doesn't go back so if you run multiple with say `tox -p all`; the second one will fail because its trying to run in the tests folder!

File: src/pybert/solvers/simbeor/__init__.py (hunk 10 above)
Comment (jdpatt) [reply src/pybert/solvers/simbeor/__init__.py:36]: Correct! That was actually python 2 to 3 change!  As long as we aren't needing to pass anything, no need to explicitly do it!

https://docs.python.org/3/library/functions.html#super

File: src/pybert/pybert.py (hunk 9 above)
Comment (jdpatt) [reply src/pybert/pybert.py:641]: Same story here, that is no longer required as of python 3.0!  https://docs.python.org/3/library/functions.html#super

File: src/pybert/pybert.py (hunk 8 above)
Comment (jdpatt) [reply src/pybert/pybert.py:350]: That is no longer required as of python 3.0 but was required when you were running python 2.7!
 https://docs.python.org/3/library/functions.html#super

File: .gitignore [hunk 13]
```diff
@@ -1,14 +1,115 @@
-*.swp
+# Byte-compiled / optimized / DLL files
```
[the remark this answers is not in the document]
Comment (jdpatt) [reply .gitignore:1]: yep, will try to clean up the diff!

File: MANIFEST.in (hunk 1 above)
Comment (jdpatt) [reply MANIFEST.in:9]: I think that was a copy paste error on my part; will add back!

File: MANIFEST.in [hunk 14]
```diff
@@ -1,10 +1,4 @@
 include README.md
 include LICENSE
-include requirements.txt
-include pybert/images/icon.png
-recursive-include docs/
-recursive-include tests/
```
[the remark this answers is not in the document]
Comment (jdpatt) [reply MANIFEST.in:6]: My understanding which might be wrong, when a user downloads your wheel or conda package that it would include the tests folder but none of the other package infrastructure like our pre-commit file or other.  So if you want to develop, we want to force them to clone this repo.  That is also less files/space in the download for install time? (Again, I could be completely wrong.)

File: README.md (hunk 2 above)
Comment (jdpatt) [reply README.md:38]: Another copy paste error from develop, will fix.

**Comment by jdpatt** [comment]
> Hi @jdpatt ,
> 
> Wow! That was a significant amount of work; thanks! I do have some questions that I'd like to discuss with you before we merge.
> 
> Thanks, -db

Thankfully most of these were automated finds/fixes from `isort`, `black` and `pylint`!!  I will hopefully push an update this weekend or next week!

File: MANIFEST.in (hunk 14 above)
[the remark this answers is not in the document]
Comment (capn-freako) [reply MANIFEST.in:6]: Okay, makes sense; thanks!

File: README.md (hunk 2 above)
Comment (jdpatt) [reply README.md]: fixed

File: MANIFEST.in (hunk 1 above)
Comment (jdpatt) [reply MANIFEST.in]: fixed

**Comment by jdpatt** [comment]
Apparently, the magic is merge master back into these branches if you work ahead on a few things and github will update the diff render when you do that prior to you merging the branch back to master!

**Comment by jdpatt** [comment]
I need to make the sim/simulation to models/bert.py a git rename not a deletion and addition to keep all of our git history. I think that is my last remaining fix here just ran out of time for the night.  Will comment when I have that complete.

File: src/pybert/pybert.py (hunk 7 above)
Comment (jdpatt) [reply src/pybert/pybert.py]: I initially missed this but now corrected.

**Comment by jdpatt** [comment]
Actually, it looks to already have tracked thru the two renames pybert_cntrl.py -> simulation.py -> bert.py and have all the history before that.

Screen Shot 2022-11-12 at 4 58 50 PM

**Comment by capn-freako** [comment]
> Actually, it looks to already have tracked thru the two renames pybert_cntrl.py -> simulation.py -> bert.py and have all the history before that.
> 
> Screen Shot 2022-11-12 at 4 58 50 PM

Gotta love Git! :)

**Review by capn-freako — APPROVED** [review]
Thanks, David!
I really appreciate all this time you've invested. :)
-db

**Comment by capn-freako** [comment]
Hi @jdpatt ,

Would you, please, bump the `Chaco` dependency to 5.1.0 and push an update?

Thanks,
-db

```
ERROR: Cannot install pybert==3.5.7 and pyibis-ami==3.5.8 because these package versions have conflicting dependencies.

The conflict is caused by:
    pybert 3.5.7 depends on chaco==5.0.0
    pyibis-ami 3.5.8 depends on chaco==5.1.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict
```

**Comment by jdpatt** [comment]
I still see `5.0.0` in both PyAMI and PyBERT and versions at `v3.5.7` in the pyproject.toml.  Have you made more changes that might not be pushed to a remote branch or is this in the conda-deps?

I did update the submodule of PyAMI up to the latest on master.

**Comment by capn-freako** [comment]
Hi @jdpatt ,

After squashing/merging here, I pulled the new `master` to my own local sandbox and ran a `tox`.
Everything passed, except the `py39` and `py310`, which is expected, as I don't have those Python versions installed, yet.

I think we're getting this _GitHub_ thing figured out! ;-)

Cheers,
-db

**Comment by jdpatt** [comment]
Yay, awesome to hear!  You'll see a CLI coming next!
